src/main/java/org/cactoos/scalar/ItemAt.java

```diff
@@ -170,12 +145,12 @@ public ItemAt(
     /**
      * Ctor.
      *
-     * @param iterator Iterator
+     * @param iterable Iterable
      * @param position Position
      */
-    public ItemAt(final Iterator<T> iterator, final int position) {
+    public ItemAt(final Iterable<T> iterable, final int position) {
         this(
-            iterator,
+            iterable.iterator(),
```

Comment:
@mazdack please add a `@todo` to continue the work on this class since there are many constructor with `Iterator` left